### ISSUE TITLE
Default project template results in build warnings with 5.0-rc1 SDK

### ISSUE THREAD

**Issue by eerhardt** [dotnet/aspnetcore#25077]
### Describe the bug
When I create a default Blazor WASM project, and build I get warnings by default.

### To Reproduce
1. Install the latest 5.0 SDK - 5.0.100-rc.1.20420.7
2. `dotnet new blazorwasm`
3. Open the project in VS and build the project

#### Expected results
Clean build

#### Actual results
```
Severity	Code	Description	Project	File	Line	Suppression State	Detail Description
Warning	CA1014	Mark assemblies with CLSCompliant	Blazor5Test	F:\DotNetTest\Blazor5Test\CSC	1	Active
Warning	CA1052	Type 'Program' is a static holder type but is neither static nor NotInheritable	Blazor5Test	F:\DotNetTest\Blazor5Test\Program.cs	13	Active
Warning	CA2007	Consider calling ConfigureAwait on the awaited task	Blazor5Test	F:\DotNetTest\Blazor5Test\Program.cs	22	Active	When an asynchronous method awaits a Task directly, continuation occurs in the same thread that created the task. Consider calling Task.ConfigureAwait(Boolean) to signal your intention for continuation. Call ConfigureAwait(false) on the task to schedule continuations to the thread pool, thereby avoiding a deadlock on the UI thread. Passing false is a good option for app-independent libraries. Calling ConfigureAwait(true) on the task has the same behavior as not explicitly calling ConfigureAwait. By explicitly calling this method, you're letting readers know you intentionally want to perform the continuation on the original synchronization context.
```

### Further technical details
- ASP.NET Core version
- Include the output of `dotnet --info`

```
.NET SDK (reflecting any global.json):
 Version:   5.0.100-rc.1.20420.7
 Commit:    3232fe7ac1

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19041
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\5.0.100-rc.1.20420.7\

Host (useful for support):
  Version: 5.0.0-rc.1.20417.14
  Commit:  907f7da59b
```
- The IDE (VS / VS Code/ VS4Mac) you're running on, and it's version

```
Microsoft Visual Studio Enterprise 2019 Preview
Version 16.8.0 Preview 1.0
VisualStudio.16.Preview/16.8.0-pre.1.0+30404.54
```

**Comment by davidfowl** [comment]
What are these warnings? Our new analyzers?

**Comment by eerhardt** [comment]
> What are these warnings? Our new analyzers?

I don't know. I'm confused at why I'm seeing them as well.

**Comment by pranavkm** [comment]
@mavasani do you happen to know what triggers these warnings?

**Comment by pranavkm** [comment]
These warnings also seem to appear in other project templates:

### API

```
1>D:\temp\demo\Program.cs(12,18,12,25): warning CA1052: Type 'Program' is a static holder type but is neither static nor NotInheritable
1>D:\temp\demo\Controllers\WeatherForecastController.cs(36,32,36,49): warning CA5394: Random is an insecure random number generator. Use cryptographically secure random number generators when randomness is required for security.
1>D:\temp\demo\Controllers\WeatherForecastController.cs(37,37,37,63): warning CA5394: Random is an insecure random number generator. Use cryptographically secure random number generators when randomness is required for security.
1>D:\temp\demo\Controllers\WeatherForecastController.cs(22,34,22,52): warning CA1823: Unused field 'scopeRequiredByApi'
1>CSC : warning CA1014: Mark assemblies with CLSCompliant
```

### MVC \ webapp

```
1>D:\temp\demo\Program.cs(12,18,12,25): warning CA1052: Type 'Program' is a static holder type but is neither static nor NotInheritable
1>CSC : warning CA1014: Mark assemblies with CLSCompliant
```

### Console

```
1>CSC : warning CA1014: Mark assemblies with CLSCompliant
1>D:\temp\demo\Program.cs(9,31,9,45): warning CA1303: Method 'void Program.Main(string[] args)' passes a literal string as parameter 'value' of a call to 'void Console.WriteLine(string? value)'. Retrieve the following string(s) from a resource table instead: "Hello World!".
1>D:\temp\demo\Program.cs(7,35,7,39): warning CA1801: Parameter args of method Main is never used. Remove the parameter or use it in the method body.
```

Are there warnings intentional?

**Comment by mavasani** [comment]
These are not intentional. Adding @jmarolf - I don't see these warnings when explicitly referencing `Microsoft.CodeAnalysis.NetAnalyzers` NuGet package. Did something break in the insertion? Could this be related to not setting the default ruleset? I am investigating as well..

**Comment by eerhardt** [comment]
Thanks @mavasani. Offline @jmarolf mentioned this is an internally tracked work item that has already had a fix checked in for it.

https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1176424

For Blazor WASM, I think our template should change from `public class Program` to `public static class Program`. That way even if this analyzer is run, it won't give this warning. It is the "more correct" thing to do IMO.

**Comment by mavasani** [comment]
@eerhardt Jon's fix to https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1176424 actually uncovered another issue that caused this behavior (you will not repro the current issue without that fix). I have another VS PR out that should fix this particular regression: https://devdiv.visualstudio.com/DevDiv/_git/VS/pullrequest/269145

**Comment by mavasani** [comment]
> For Blazor WASM, I think our template should change from public class Program to public static class Program. That way even if this analyzer is run, it won't give this warning. It is the "more correct" thing to do IMO.

Yep, that could be a follow-up change on Blazor WASM side.

**Comment by mavasani** [comment]
Fix merged into VS2019 16.8 P2 with https://devdiv.visualstudio.com/DevDiv/_git/VS/pullrequest/269157

**Comment by pranavkm** [comment]
@mavasani would we continue to see the warning for a `static Program` in 16.8 P2?

**Comment by jmarolf** [comment]
no 16.8 P2 should not warn with @mavasani's change

**Comment by pranavkm** [comment]
Given there isn't much to gain making Program.cs a static type (across our templates), we'll be fine to leave it be. Closing this. Thanks everyone for the quick turn around on this 🚀

**Comment by davidfowl** [comment]
I see a crapload of warnings after installing the latest build of RC1.  I'm on a build later than 16.8 P2 and the warnings flash on and off in the IDE. Is this expected?

**Comment by mavasani** [comment]
Fix got merged into VS2019 16.8 P2 today afternoon and the backport PR to main branch is currently running tests. I hope to have it merged soon so tomorrow’s dogfood build should have the fix.

**Comment by davidfowl** [comment]
Thanks @mavasani 

See these warnings as well 😄 

```
2>C:\Users\david\source\repos\WebApplication2\APIPlayground\Program.cs(27,1,27,6): warning CA1812: <Program>$ is an internal class that is apparently never instantiated. If so, remove the code from the assembly. If this class is intended to contain only static members, make it static (Shared in Visual Basic).
2>CSC : warning AD0001: Analyzer 'Microsoft.CodeQuality.Analyzers.ApiDesignGuidelines.IdentifiersShouldNotContainTypeNames' threw an exception of type 'System.IndexOutOfRangeException' with message 'Index was outside the bounds of the array.'.
```
